src/macports1.0/macports.tcl

```diff
@@ -548,6 +548,54 @@ proc macports::_is_valid_developer_dir {dir} {
     return 1
 }
 
+##
+# Obtain variables describing the current OS version and return them in a list
+# suitable for lassign(n).
+#
+# A typical use-case for this function is lassign(n)ing its result as follows:
+#
+#   lassign [macports::_get_os_version] \
+#       os_arch \
+#       os_version \
+#       os_major \
+#       os_minor \
+#       os_platform \
+#       os_endian \
+#       macosx_version
+#
+# @return A list containing the OS architecture, complete OS version, OS major
+#         and minor versions, the OS platform, the OS endianess, and the macOS
+#         version (empty on non-macOS systems)
+proc macports::_get_os_version {} {
+    # set up platform info variables
+    set os_arch $tcl_platform(machine)
+    if {$os_arch eq "Power Macintosh"} {
+        set os_arch "powerpc"
+    }
+    if {$os_arch in {"i586" "i686" "x86_64"}} {
+        set os_arch "i386"
+    }
+
+    set os_version $tcl_platform(osVersion)
```

Comment:
This one is giving build error. Need to take a look.
https://travis-ci.org/macports/macports-base/builds/327000161#L2308